internal/model/database.go

```diff
@@ -0,0 +1,281 @@
+	//
+	// Arguments:
+	//
+	// - loc: loc is the location provider used to instantiate the network
+	//
+	// Returns either a database network instance or an error
+	CreateNetwork(loc LocationProvider) (*DatabaseNetwork, error)
+
+	// CreateOrUpdateURL will create a new URL entry to the urls table if it doesn't
+	// exists, otherwise it will update the category code of the one already in
+	// there.
+	//
+	// Arguments:
+	//
+	// - urlStr is the URL string to create or update
+	//
+	// - categoryCode is the category code to update
+	//
+	// - countryCode is the country code to update
+	//
+	// Returns either the new URL id or an error
+	CreateOrUpdateURL(urlStr string, categoryCode string, countryCode string) (int64, error)
+
+	// CreateResult writes the Result to the database a returns a pointer
+	// to the Result
+	//
+	// Arguments:
+	//
+	// - homePath is the home directory path to make the results directory
+	//
+	// - testGroupName is used to annotate the database the result
+	//
+	// - networkID is the id of the underlying network
+	//
+	// Returns either a database result instance or an error
+	CreateResult(homePath string, testGroupName string, networkID int64) (*DatabaseResult, error)
+
+	// UpdateUploadedStatus will check if all the measurements inside of a given result set have been
+	// uploaded and if so will set the is_uploaded flag to true
+	//
+	// Arguments:
+	//
+	// - result is the database result to update
+	//
+	// Returns a non-nil error if update failed
+	UpdateUploadedStatus(result *DatabaseResult) error
+
+	// Finished marks the result as done and sets the runtime
+	//
+	// Arguments:
+	//
+	// - result is the database result to mark as done
+	//
+	// Returns a non-nil error if result could not be marked as done
+	Finished(result *DatabaseResult) error
+
+	// DeleteResult will delete a particular result and the relative measurement on
+	// disk.
+	//
+	// Arguments:
+	//
+	// - resultID is the id of the database result to be deleted
+	//
+	// Returns a non-nil error if result could not be deleted
+	DeleteResult(resultID int64) error
+
+	// CreateMeasurement writes the measurement to the database a returns a pointer
+	// to the Measurement
+	//
+	// Arguments:
+	//
+	// - reportID is the report id to annotate
+	//
+	// - testName is the experiment name to use
+	//
+	// - measurementDir is the measurement directory path
+	//
+	// - resultID is the result id to annotate
+	//
+	// - urlID is the id of the URL input
+	//
+	// Returns either a database measurement or an error
+	CreateMeasurement(reportID sql.NullString, testName string, measurementDir string, idx int,
+		resultID int64, urlID sql.NullInt64) (*DatabaseMeasurement, error)
+
+	// AddTestKeys writes the summary to the measurement
+	//
+	// Arguments:
+	//
+	// - msmt is the database measurement to update
+	//
+	// - tk is the testkeys
+	//
+	// Returns a non-nil error if measurement update failed
+	AddTestKeys(msmt *DatabaseMeasurement, tk interface{}) error
```

Comment:
```suggestion
	AddTestKeys(msmt *DatabaseMeasurement, tk any) error
```

Since go1.18, we should use `any` rather than `interface{}`.